src/common/com/intellij/plugins/haxe/lang/psi/HaxeClassResolveResult.java

```diff
@@ -95,17 +98,28 @@ public static HaxeClassResolveResult create(@Nullable HaxeClass aClass, HaxeGene
           LOG.debug(debugNestCountForCreate + "  Not adding specialization for " + aClass.getName() + "<" + componentName.getName() + ">. No specialized type found.");
         }
       }
+      // END constraint block.
 
-      for (HaxeType haxeType : aClass.getHaxeExtendsList()) {
-        final HaxeClassResolveResult result = create(HaxeResolveUtil.tryResolveClassByQName(haxeType));
-        result.specializeByParameters(haxeType.getTypeParam());
-        LOG.debug(debugNestCountForCreate + "  Adding (extends) specialization for " + aClass.getName() + "<" + haxeType.getName() + "> -> " + result.getSpecialization().debugDump("    "));
-        resolveResult.merge(result.getSpecialization());
-      }
-      for (HaxeType haxeType : aClass.getHaxeImplementsList()) {
-        final HaxeClassResolveResult result = create(HaxeResolveUtil.tryResolveClassByQName(haxeType));
-        result.specializeByParameters(haxeType.getTypeParam());
-        LOG.debug(debugNestCountForCreate + "  Adding (implements) specialization for " + aClass.getName() + "<" + haxeType.getName() + "> -> " + result.getSpecialization().debugDump("    "));
+      // Load the specialization with sub-class parameters.
```

Comment:
The changes in the following block are the central changes to resolving chained generic types correctly.  While we used to not be able to resolve superclasses, we now create a list of resolved arguments from the instantiation, and pass those down.  It required that we figure out all of the real types that were represented by generic variables before resolving the superclass, and then creating a new list (generateParameterList()) instead of using the type parameters from the declaration (which the old code did).